libs/openFrameworks/utils/ofThreadChannel.h

```diff
@@ -1,28 +1,73 @@
 #pragma once
+
+
 #include <mutex>
 #include <queue>
 #include <condition_variable>
 #include "ofUtils.h"
 
-/// Communication channel between different threads
-/// allows for multithreaded programming without using
-/// mutexes or other synchronization structures.
-/// Instead of using some shared memory send a message
-/// through a channel whenever one thread needs to send
-/// some data to another.
-/// Usually a second channel can be used to send the
-/// processed data back to the first thread.
+
+/// \brief Safely send data between threads without additional synchronization.
+///
+/// ofThreadChannel makes it easy to safely and efficiently share data between
+/// threads without the need for shared memory, mutexes, or other
+/// synchronization techniques. Additionally, ofThreadChannel employs a
+/// signalling system that allows receiving threads to sleep until new data
+/// arrives or the ofThreadChannel is closed.
+///
+/// A single ofThreadChannel class is desgined for one-way communication. In
+/// most cases an additional ofThreadChannel can be used for two-way
+/// communication.
+///
+/// Data is sent through the ofThreadChannel in a FIFO (first in, first out)
+/// order, guaranteeing that that data will be received in the same order that
+/// it was sent.
+///
+/// If multiple threads attempt to send data using the same ofThreadChannel, the
+/// send method will block the calling thread until it is free. Sending and
+/// receiving data from the same thread may result in deadlocks.
+///
```

Comment:
i don't think this could happen, a thread could get blocked for ever waiting on data if there's no data available but i think a deadlock is impossible?